plugins-extra/query_enhancements/public/query_assist/components/query_assist_input.tsx

```diff
@@ -0,0 +1,75 @@
+import { EuiFieldText, EuiIcon, EuiOutsideClickDetector, EuiPortal } from '@elastic/eui';
+import React, { useMemo, useState } from 'react';
+import {
+  PersistedLog,
+  QuerySuggestionTypes,
+  SuggestionsComponent,
+} from '../../../../../src/plugins/data/public';
+import assistantLogo from '../../assets/query_assist_logo.svg';
+
+interface QueryAssistInputProps {
+  inputRef: React.RefObject<HTMLInputElement>;
+  persistedLog: PersistedLog;
+  initialValue?: string;
+  selectedIndex?: string;
+  previousQuestion?: string;
+}
+
+export const QueryAssistInput: React.FC<QueryAssistInputProps> = (props) => {
```

Comment:
we are just using recent suggestions right now right?

i'd imagine its similar to assistant dashboards the initial will keep track of the suggestions as they get updated for reference? https://github.com/opensearch-project/dashboards-assistant/blob/main/public/services/incontext_insight/incontext_insight_registry.ts#L63

Reply:
or is the plan that we will just persist it

Reply:
currently just recent suggestions, maybe plus some hard-coded suggestions for sample data.

i didn't really understand the difference,

> initial will keep track of the suggestions as they get updated for reference

is this for dedup purpose, or when user edits a question and submit, the recent suggestions gets edited instead of adding a new one?